lib/nanoc/base/compilation/compiler.rb

```diff
@@ -233,8 +237,10 @@ def compile_rep(rep)
       if can_reuse_content_for_rep?(rep)
         Nanoc::Int::NotificationCenter.post(:cached_content_used, rep)
         rep.snapshot_contents = compiled_content_cache[rep]
+        rep.status = :created
```

Comment:
Hmm, this `:created` is incorrect. Some context:

If `#can_reuse_content_for_rep?` returns `true`, then that means that the compiled content (which was cached in the previous compilation run) can be reused without actually recompiling the item. This is used when an outdated item includes content from an item that is not outdated, and whose content can thus be reused without recompiling.

In the `ItemRepWriter`, Nanoc determines whether the resulting file exists already (`is_created`), is modified (`is_modified`), or neither. (See [these `ItemRepWriter` lines](https://github.com/nanoc/nanoc/blob/master/lib/nanoc/base/services/item_rep_writer.rb#L32-L33).)

It’d be useful to make use of the `ItemRepWriter` modified/created logic that already exists. Creating a snapshot for which a route is defined will make the item rep writer write out the compiled content. The `:last` snapshot, containing the compiled content after the entire rule is executed, is trigged by the line `executor.snapshot(rep, :last)` in the `#recalculate_content_for_rep` method below.

Perhaps the `ItemRepWriter` could update the item rep to set `item_rep.modified = is_modified`, with `is_modified` being whatever `ItemRepWriter` already calculates. (I don’t think there’s a need for a similar `created` attribute.) The advantage of this is that *only* `ItemRepWriter` determines whether content is deemed changed. In the following cases, ItemRepWriter will treat the content as not modified:

* An item that is recompiled (`#can_reuse_content_for_rep?` returns `false`), but whose content turns out to be identical after compilation

* An item whose content is reused (`#can_reuse_content_for_rep?` returns `true`)

* An item that’s not even considered for compilation (not considered outdated)